docs/src/apimanual.md

```diff
@@ -56,39 +56,44 @@ The current function types are:
 * **scalar-valued quadratic**: ``\frac{1}{2} x^T Q x + a^T x + b``, where ``Q`` is a symmetric matrix, ``a`` is a vector, and ``b`` is a constant
 * **vector-valued quadratic**: a vector of scalar-valued quadratic expressions
 
-In a future version, MOI could be extended to cover functions defined by evaluation oracles (e.g., for nonlinear derivative-based optimization).
+Extensions for nonlinear programming are present but not yet well documented.
 
 MOI defines some commonly used sets, but the interface is extensible to other sets recognized by the solver.
 [Describe currently supported sets.]
 
-## Instances
+## The `ModelLike` and `AbstractOptimizer` APIs
 
-An **Instance** ([`AbstractInstance`](@ref MathOptInterface.AbstractInstance)) is a representation of a concrete instance of an optimization problem, i.e., with all data specified.  Instances are either **standalone instances** or **solver instances**:
+The most significant part of MOI is the definition of the **model API** that is
+used to specify an instance of an optimization problem (e.g., by adding
+variables and constraints). Objects that implement the model API should inherit
```

Comment:
"Objects that implement the model API" -> "Objects that implement **parts of** the model API" ? Or we consider that if an object does not implement something and say `can...` -> `false` then it is a complete implementation ?

Reply:
We should definitely add a discussion in the documentation of the different ways to implement the model API, but for the purposes of this sentence, I consider copy-only a valid implementation of the model API.

Reply:
Yes, I think that if `can...` return false, it should be considered a valid implementation